File: MyBudget-back/src/main/java/com/example/mybudget/services/AccountService.java
package com.example.mybudget.services;

import com.example.mybudget.models.dto.Account;
import com.example.mybudget.models.requests.AccountRequest;

import java.math.BigDecimal;
import java.util.List;

public interface AccountService {
    Account create(AccountRequest request);

    List<Account> getAll();

    void deleteAllData();

    void importData();

    BigDecimal getAvailableBalance(String currency);
}

File: MyBudget-front/src/app/services/account.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AccountRequest } from '../models/account-request';
import { Account } from '../models/account';

@Injectable({
  providedIn: 'root',
})
export class AccountService {
  private apiUrl = 'http://localhost:8080/accounts';

  constructor(private http: HttpClient) {}

  getAll() {
    return this.http.get<any>(this.apiUrl);
  }

  create(request: AccountRequest) {
    return this.http.post<Account>(this.apiUrl, request);
  }

  getAvailableBalance(currency: String) {
    return this.http.get<number>(`${this.apiUrl}/available/${currency}`);
  }

  deleteAllData() {
    return this.http.delete(this.apiUrl);
  }
}
